modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -381,6 +381,15 @@ impl LightningClientModule {
             return Err(SendPaymentError::InvoiceExpired);
         }
 
+        let federation_currency: Currency = self.cfg.network.into();
+        let invoice_currency = invoice.currency();
+        if federation_currency != invoice_currency {
```

Comment:
Please do not assign extra variables here, compare them directly for consistency with lnv2.